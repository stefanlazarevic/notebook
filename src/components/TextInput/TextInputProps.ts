import PropTypes, { InferProps, Validator } from "prop-types";

export const TextInputPropTypes = {
	/**
	 *
	 */
	id: PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	testid: PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	className: PropTypes.string as Validator<string | undefined>,
	/**
	 * @default ""
	 */
	value: PropTypes.string as Validator<string | undefined>,
	/**
	 * @default ""
	 */
	placeholder: PropTypes.string as Validator<string | undefined>,
	/**
	 * @default false
	 */
	disabled: PropTypes.bool as Validator<boolean | undefined>,
	/**
	 * @default false
	 */
	invalid: PropTypes.bool as Validator<boolean | undefined>,
	/**
	 *
	 */
	"aria-labelledby": PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	onChange: PropTypes.func as Validator<
		(event: React.SyntheticEvent<HTMLInputElement>, value: string) => void | undefined
	>,
	/**
	 *
	 */
	onFocus: PropTypes.func as Validator<(event: React.FocusEvent<HTMLInputElement>) => void | undefined>,
	/**
	 *
	 */
	onBlur: PropTypes.func as Validator<(event: React.FocusEvent<HTMLInputElement>) => void | undefined>,
};

export type TextInputProps = InferProps<typeof TextInputPropTypes>;
