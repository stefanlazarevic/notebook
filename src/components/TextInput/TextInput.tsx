import React, { useMemo } from "react";

import "./TextInput.css";

import { TextInputPropTypes, TextInputProps } from "./TextInputProps";

function TextInput(props: TextInputProps) {
	const className = useMemo(() => {
		return `TextInput ${props.className || ""}`.trim();
	}, [props.className]);

	function onChange(event: React.ChangeEvent<HTMLInputElement>) {
		if (typeof props.onChange === "function" && !props.disabled) {
			props.onChange(event, event.target.value);
		}
	}

	return (
		<input
			id={props.id}
			data-testid={props.testid}
			type="text"
			className={className}
			value={props.value}
			onChange={onChange}
			onFocus={props.onFocus}
			onBlur={props.onBlur}
			disabled={props.disabled}
			placeholder={props.placeholder}
			aria-invalid={props.invalid}
			aria-labelledby={props["aria-labelledby"]}
		/>
	);
}

TextInput.propTypes = TextInputPropTypes;

TextInput.defaultProps = {
	value: "",
	placeholder: "",
	disabled: false,
	invalid: false,
};

TextInput.displayName = "TextInput";

export default TextInput;
