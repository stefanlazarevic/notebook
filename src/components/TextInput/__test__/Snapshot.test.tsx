import React from "react";
import { render, cleanup } from "@testing-library/react";
import TextInput from "../index";

afterEach(cleanup);

it("Подразумеванa својства се подударају са последњим снимком.", () => {
	const { asFragment } = render(<TextInput />);

	expect(asFragment()).toMatchSnapshot();
});

it("Сва подешена својства се подударају са последњим снимком.", () => {
	const { asFragment } = render(
		<TextInput
			id="id"
			testid="test-id"
			aria-labelledby="test-label"
			value="test"
			className="CustomTextInput"
			disabled={false}
			invalid={true}
			placeholder="Test TextInput"
		/>
	);

	expect(asFragment()).toMatchSnapshot();
});
