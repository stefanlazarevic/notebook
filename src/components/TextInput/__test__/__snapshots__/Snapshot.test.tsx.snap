// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Подразумеванa својства се подударају са последњим снимком. 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    class="TextInput"
    placeholder=""
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`Сва подешена својства се подударају са последњим снимком. 1`] = `
<DocumentFragment>
  <input
    aria-invalid="true"
    aria-labelledby="test-label"
    class="TextInput CustomTextInput"
    data-testid="test-id"
    id="id"
    placeholder="Test TextInput"
    type="text"
    value="test"
  />
</DocumentFragment>
`;
